Gems/Maestro/Code/Source/Components/EditorSequenceComponent.cpp

```diff
@@ -234,10 +249,50 @@ namespace Maestro
     ///////////////////////////////////////////////////////////////////////////////////////////////////////
     void EditorSequenceComponent::RemoveEntityToAnimate(AZ::EntityId removedEntityId)
     {
+        if (!GetEntity())
+        {
+            // This check removes ambiguous warning in Component::GetEntityId()
+            // while destroying an instance when sanitizing a prefab DOM :
+            // entities are not activated, components have no pointers to parent entities,
+            // buses are not connected -> no need to try fetching owner EntityId from m_sequence.
+            return;
+        }
+
         const Maestro::SequenceAgentEventBusId ebusId(GetEntityId(), removedEntityId);
 
         // Notify the SequenceAgentComponent that we're disconnecting from it
         Maestro::SequenceAgentComponentRequestBus::Event(ebusId, &Maestro::SequenceAgentComponentRequestBus::Events::DisconnectSequence);
+
+        // TODO (AnimSequence linking):
+        // Consider destroying the now ambiguous CAnimAzEntityNode right here, 
+        // please see the closely related comment with the same "TODO (AnimSequence linking)" tag
+        // in the component dtor above, where this method is called directly.
+        // Removing the node disables any following ambiguous bus calls to the
+        // EditorSequenceAgentComponent::DisconnectSequence() in this component,
+        // as the corresponding agent component is now detached and "dead"
+        // (please see comments in the EditorSequenceAgentComponent::DisconnectSequence()).
+        // Although it seems that destroying the now ambiguous node here would be architecturally right,
+        // this may interfere with the current implementation of the governing code.
+        // In ideal world the governing code calling RemoveEntityToAnimate() is then to call the AddEntityToAnimate()
+        // in case the Sequence <-> SequenceAgent link is further used.
+        #if 0
```

Comment:
Will this be changed in a different PR? any reason to keep it disabled?

Reply:
@lsemp3d @guillaume-haerinck 
The final fix for the issue does not need removing the CAnimAzEntityNode, so I deleted most of the above comments, leaving a TODO in a single place - because I believe this architectural inconsistency deserves mentioning.

The PR description above updated.
Code is ready for merging.
Thanks for your patience.